airflow/jobs.py

```diff
@@ -1024,35 +1024,39 @@ def _execute_task_instances(self,
                 # Arbitrary:
                 # If queued outside of a pool, trigger no more than
                 # non_pooled_task_slot_count per run
-                open_slots = conf.getint('core', 'non_pooled_task_slot_count')
+                slots = conf.getint('core', 'non_pooled_task_slot_count')
+                open_slots = slots
+                pool_name = 'default_pool'
             else:
+                pool_name = pool
                 if pool not in pools:
                     self.logger.warning(
                         "Tasks using non-existent pool '%s' will not be scheduled",
                         pool
                     )
-                    open_slots = 0
+                    continue
                 else:
                     open_slots = pools[pool].open_slots(session=session)
+                    slots = pools[pool].slots
 
             num_queued = len(task_instances)
             self.logger.info("Figuring out tasks to run in Pool(name={pool}) "
                              "with {open_slots} open slots and {num_queued} "
                              "task instances in queue".format(**locals()))
 
-            if open_slots <= 0:
-                continue
-
             priority_sorted_task_instances = sorted(
                 task_instances, key=lambda ti: (-ti.priority_weight, ti.execution_date))
 
             # DAG IDs with running tasks that equal the concurrency limit of the dag
             dag_id_to_possibly_running_task_count = {}
 
-            for task_instance in priority_sorted_task_instances:
+            # Number of tasks that cannot be scheduled because of no open slot in pool
+            num_starving_tasks = 0
+            for current_index, task_instance in enumerate(priority_sorted_task_instances):
                 if open_slots <= 0:
                     self.logger.info("No more slots free")
                     # Can't schedule any more since there are no more open slots.
+                    num_starving_tasks = len(priority_sorted_task_instances) - current_index
```

Comment:
```suggestion
                    num_starving_tasks += 1
```

Would this be simpler?

Reply:
The reason we need this is because we break the loop immediately when there are no more `open_slots`.

Reply:
Ohhh, I see. Thanks